### PR TITLE
80772 get more than 10 commpkgs

### PR DESCRIPTION
**Pull request by bratli**
DevOps#80772

Commpkg search has itemsPerPage and currentPage to be able to get more than 10 commpkgs in search

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.WebApi/Controllers/Scope/ScopeController.cs [hunk 1]
```diff
@@ -26,17 +26,21 @@ public class ScopeController : ControllerBase
         /// <param name="plant"></param>
         /// <param name="projectName"></param>
         /// <param name="startsWithCommPkgNo"></param>
+        /// <param name="itemsPerPage"></param>
+        /// <param name="currentPage"></param>
         /// <returns>All ProCoSys commpkgs that match the search parameters</returns>
         [Authorize(Roles = Permissions.COMMPKG_READ)]
         [HttpGet("CommPkgs")]
-        public async Task<ActionResult<List<ProCoSysCommPkgDto>>> GetCommPkgsInProject(
+        public async Task<ActionResult<ProCoSysCommPkgSearchDto>> GetCommPkgsInProject(
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.WebApi/Controllers/Scope/ScopeController.cs:34]: NBNB! This is a breaking change for client!! How will you solve this for existing client?

Comment (bratli) [reply src/Equinor.ProCoSys.IPO.WebApi/Controllers/Scope/ScopeController.cs:34]: Will merge this PR at the same time as Sverre takes out UI was my first thought. Now I see that that will result in QP not working if UI is put there a long time before being taken to prod.

What is the procedure for cases like this?

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Scope/ScopeControllerTests.cs [hunk 2]
```diff
@@ -11,16 +11,17 @@ public class ScopeControllerTests : ScopeControllerTestsBase
...
                 UserType.Viewer,
                 TestFactory.PlantWithAccess,
                 TestFactory.ProjectWithAccess,
                 "CommPkgNo");
 
             // Assert
-            var commPkg1 = commPkgs.First();
-            var commPkg2 = commPkgs.Last();
-            Assert.AreEqual(2, commPkgs.Count);
+            var commPkg1 = commPkgSearchResult.CommPkgs.First();
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Scope/ScopeControllerTests.cs:21]: Changes as this in an integration test "proof" the breaking change mentioned in controller.
We should discuss how to solve it ..

**Review by janid1967 — CHANGES_REQUESTED** [review]
must solve the breaking change somehow

File: src/Equinor.ProCoSys.IPO.WebApi/Controllers/Scope/ScopeController.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.IPO.WebApi/Controllers/Scope/ScopeController.cs:34]: It depends .... we must discuss

File: src/Equinor.ProCoSys.IPO.Query/GetCommPkgsInProject/GetCommPkgsInProjectOldQuery.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using Equinor.ProCoSys.IPO.Domain;
+using MediatR;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Query.GetCommPkgsInProject
+{
+    public class GetCommPkgsInProjectOldQuery : IProjectRequest, IRequest<Result<IList<ProCoSysCommPkgDto>>>
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Query/GetCommPkgsInProject/GetCommPkgsInProjectOldQuery.cs:8]: remove class

File: src/Equinor.ProCoSys.IPO.Query/GetCommPkgsInProject/GetCommPkgsInProjectOldQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.IPO.Domain;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.CommPkg;
+using MediatR;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Query.GetCommPkgsInProject
+{
+    public class GetCommPkgsInProjectOldQueryHandler : IRequestHandler<GetCommPkgsInProjectOldQuery, Result<IList<ProCoSysCommPkgDto>>>
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Query/GetCommPkgsInProject/GetCommPkgsInProjectOldQueryHandler.cs:12]: remove class

File: src/Equinor.ProCoSys.IPO.WebApi/Controllers/Scope/ScopeController.cs [hunk 5]
```diff
@@ -29,14 +54,15 @@ public class ScopeController : ControllerBase
...
-        public async Task<ActionResult<List<ProCoSysCommPkgDto>>> GetCommPkgsInProject(
+        public async Task<ActionResult<IList<ProCoSysCommPkgDto>>> GetCommPkgsInProject(
             [FromHeader(Name = CurrentPlantMiddleware.PlantHeader)]
             [Required]
             string plant,
             [FromQuery] string projectName,
             [FromQuery] string startsWithCommPkgNo)
         {
-            var result = await _mediator.Send(new GetCommPkgsInProjectQuery(projectName, startsWithCommPkgNo));
+            var result = await _mediator.Send(new GetCommPkgsInProjectOldQuery(projectName, startsWithCommPkgNo));
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.WebApi/Controllers/Scope/ScopeController.cs:64]: revert change and return result.CommPkgs below

**Review by janid1967 — APPROVED** [review]

**Pull request merged**